### ISSUE TITLE
Trouble with optimizers in sequential

### ISSUE THREAD

**Issue by forgetableDr** [tensorflow/tfjs#683]
tfjs: 0.12.6
tfjs-core: 0.12.15
tfjs-node: 0.1.14
NodeJS latest (10.x)

Hello. everything is ok when you're using this:
`model.compile({ optimizer: 'adadelta', loss: 'meanSquaredError' })`
But adadelta this way has learning rate equal to 1 and I want to change learning rate and other params easliy.
When I try 
`model.compile({ optimizer: tf.train.adadelta(.001), loss: 'meanSquaredError' })`
it gives me the next error:

> Error: User-defined optimizer must be an instance of tf.Optimizer.

I found some setLearningRate methods which are not implemented yet or have been already deprecated. Maybe I'm missing something?

**Comment by forgetableDr** [comment]
Fixed it for now by adding opportunity to drop "optimizer" key and to add "optimizerInstance" key which binds the necessary optimizer in the training.js.

`
if (typeof config.optimizer === 'string' || config.optimizerInstance) {
    this.optimizer = config.optimizerInstance || optimizers.getOptimizer(config.optimizer)
}`

**Comment by forgetableDr** [comment]
So, I noticed that tf.train.adadelta() is an instance of the tf.Optimizer. But it is not an instance of the tfjs_core_1.Optimizer which is loading using this
`var tfjs_core_1 = require("@tensorflow/tfjs-core");`
Is this a bug or version shift? Thanks.

**Comment by tafsiri** [comment]
I think the issue could be because of a mismatch in the dependency of tfjs and tfjs-core. Do you have an explicit dependency on both in your package.json? If so I would remove the tfjs-core one. "tfjs" and "tfjs-node" should be sufficient.

**Comment by forgetableDr** [comment]
Yeah. I have updated tfjs-node@latest and tfjs@latest and removed tfjs-core and the problem is gone now. Thanks!
